frontend/src/hooks/console-metrics.ts

```diff
@@ -0,0 +1,18 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { useEffect } from 'react'
+import { getBackendUrl, postRequest } from '../resources'
+
+export enum Pages {
+  overview = 'overview-classic',
+  overviewFleet = 'overview-fleet',
+  search = 'search',
+  searchDetails = 'search-details',
+}
+
+export function usePageVisitMetricHandler(page: Pages) {
+  useEffect(() => {
+    postRequest(getBackendUrl() + `/metrics?${page}`, {})
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
`page` should be in the dependency list and the eslint-disable removed.

Reply:
added page to dependency array